packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-and-activity-cmk.ts

```diff
@@ -0,0 +1,39 @@
+import * as cdk from 'aws-cdk-lib';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as kms from 'aws-cdk-lib/aws-kms';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+class KMSStateMachine extends cdk.Stack {
+  readonly stateMachine: sfn.StateMachine;
+  readonly activity: sfn.Activity;
+  readonly kmsKey: kms.Key;
+
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    this.kmsKey = new kms.Key(this, 'Key');
+
+    this.stateMachine = new sfn.StateMachine(this, 'StateMachineWithCMKEncryptionConfiguration', {
+      stateMachineName: 'StateMachineWithCMKEncryptionConfiguration',
+      definitionBody: sfn.DefinitionBody.fromChainable(sfn.Chain.start(new sfn.Pass(this, 'Pass'))),
+      stateMachineType: sfn.StateMachineType.STANDARD,
+      kmsKey: this.kmsKey,
+      kmsDataKeyReusePeriodSeconds: cdk.Duration.seconds(75),
+    });
+
+    this.activity = new sfn.Activity(this, 'ActivityWithCMKEncryptionConfiguration', {
+      activityName: 'ActivityWithCMKEncryptionConfiguration',
+      kmsKey: this.kmsKey,
+      kmsDataKeyReusePeriodSeconds: cdk.Duration.seconds(75),
+    });
+
+  }
+}
+const app = new cdk.App();
+const stack = new KMSStateMachine(app, 'aws-stepfunctions-statemachine-and-activity-with-cmk-encryptionconfig');
+
+new IntegTest(app, 'StateMachineAndActivityWithCMKEncryptionConfiguration', {
+  testCases: [stack],
```

Comment:
This tests that Activity and StateMachine can be deployed with encryption configuration set, but doesn't verify anything beyond that. Can we verify using an CMK encrypted activity in a state machine works when created through CDK? Or have you validated that manualy for that matter.

Reply:
I did validate that manually, like you said deploying a valid CFN template is one thing, but achieving the desired behaviour is another. I wasn't sure how to setup an activity in this folder such that it starts the worker and then we start an execution. If possible I can definitely set that up, otherwise not sure if this integ test really benefits us.

Reply:
Discussed with wong-a offline to start and execution and return a successful response `SendTaskSuccess` via th sdk and then assert against the Activity input.